lib/theme_settings_manager.rb

```diff
@@ -174,9 +174,45 @@ def choices
 
   class Upload < self
     def value
-      val = super
-      Discourse.store.cdn_url(val)
+      cdn_url(super)
+    end
+
+    def default
+      upload_id = default_upload_id
+      return if upload_id.blank?
+
+      cdn_url(upload_id)
+    end
+
+    def value=(new_value)
+      if new_value.present?
+        if new_value == default
+          new_value = default_upload_id
+        else
+          upload = ::Upload.find_by(url: new_value)
+          new_value = upload.id if upload.present?
+        end
+      end
+
+      super(new_value)
+    end
+
+    private
+
+    def cdn_url(upload_id)
+      return if upload_id.blank?
+
+      upload = ::Upload.find_by_id(upload_id.to_i)
+      return if upload.blank?
+
+      Discourse.store.cdn_url(upload.url)
     end
 
+    def default_upload_id
+      theme_field = theme.theme_fields.find_by(name: @default)
```

Comment:
What happens if someone sets the default to the name of a non-upload theme_field (e.g. `head`)?

I think we need to restrict this lookup to **upload** theme fields. Maybe by checking that the type is `theme_upload_var`?

Reply:
Sure, I will check the type too